packages/web3-providers/src/rss3/index.ts

```diff
@@ -110,3 +106,22 @@ export class RSS3API implements RSS3BaseAPI.Provider, NonFungibleTokenAPI.Provid
         return createPageable(data, createIndicator(indicator))
     }
 }
+
+const getIdFromDonationURL = (url?: string) => {
+    if (!url) return
+    return url?.match(/(?<=https:\/\/gitcoin.co\/grants)\d+/g)?.[0]
+}
+
+const createCollection = (collectionResponse: RSS3BaseAPI.CollectionResponse[]): RSS3BaseAPI.Collection[] => {
+    return collectionResponse.map((collection: RSS3BaseAPI.CollectionResponse) => {
+        return {
+            ...collection,
+            title: collection?.actions?.[0]?.metadata?.title,
+            id:
+                collection?.actions?.[0]?.metadata?.id ??
+                getIdFromDonationURL(collection?.actions?.[0]?.related_urls?.[0]) ??
+                collection?.hash,
+            imageURL: collection?.actions?.[0]?.metadata?.logo ?? collection?.actions?.[0]?.metadata?.image,
```

Comment:
Please respect to typing. `collection: RSS3BaseAPI.CollectionResponse`

```suggestion
            title: collection.actions?.[0]?.metadata?.title,
            id:
                collection.actions?.[0]?.metadata?.id ??
                getIdFromDonationURL(collection.actions?.[0]?.related_urls?.[0]) ??
                collection.hash,
            imageURL: collection.actions?.[0]?.metadata?.logo ?? collection.actions?.[0]?.metadata?.image,
```